src/librbd/crypto/Utils.cc

```diff
@@ -58,6 +58,7 @@ int build_crypto(
 
   *result_crypto = BlockCrypto<EVP_CIPHER_CTX>::create(
           cct, data_cryptor, block_size, data_offset);
+  result_crypto->get()->put();
```

Comment:
A cryptor can be shared across ancestors (e.g. parent ImageCtx)

Reply:
I'm not sure I follow.  An image can have exactly one ancestor and image's data structures should outlive those of the parent.  Where do ownership ambiguities arise?

Reply:
By ancestor ImageCtx, I mean ImageCtx.parent, ImageCtx.parent.parent, etc.

The example I have in mind is this:
Take an encrypted image, clone it.
Now, load the  encryption on the clone: both ImageCtx, and ImageCtx.parent will use the same CryptoInterface.
Now, call encryption_format_thin on the cloned image. This will create a new CryptoInterface that will be used on its ImageCtx, whereas the old CryptoInterface will still be used by ImageCtx.parent

Reply:
OK, so after the encryption is loaded on the clone, ImageCtx is the owner, ImageCtx.parent is a non-owning pointer.  Then, in encryption_format_thin(), the ownership is transferred to ImageCtx.parent.  I don't see any ambiguities here.  It seems like the rule could be
- ImageCtx owns its CryptoInterface unless it is a parent of a non-formatted clone

Would something like
```
if (ictx->child && !ictx->child->is_formatted_clone) {
  destroy ictx->child->crypto;
}
```
work and allow to get rid of refcount?

Reply:
I will try that